mutiny-core/src/onchain.rs

```diff
@@ -115,19 +120,29 @@ impl<S: MutinyStorage> OnChainWallet<S> {
             .await?;
 
         // get new wallet lock for writing and apply the update
-        match self.wallet.try_write() {
-            Ok(mut wallet) => match wallet.apply_update(update) {
-                Ok(_) => wallet.commit()?,
+        loop {
```

Comment:
Should it loop forever? What about just doing a `for _ in 1..3` with a 1s sleep or something?

Reply:
changed to 10 iterations with 250ms sleep